### PR TITLE
New tutorial on Jupyter and DataHub

### PR DESCRIPTION
**Pull request by enolfc**
# Summary

A tutorial for getting a VM with Jupyter and Datahub on integrated cloud providers

---

**Related issue :**

### CODE REVIEW

**Comment by enolfc** [comment]
Pending the completion of the VM image in AppDB

**Comment by enolfc** [comment]
Related to https://github.com/EGI-Federation/fedcloud-vmi-templates/pull/40

File: content/en/users/tutorials/jupyter-datahub-virtual-machine/_index.md
Comment (sebastian-luna-valero) [inline content/en/users/tutorials/jupyter-datahub-virtual-machine/_index.md:41]:
```suggestion
security groups, otherwise you will not be able to reach the Jupyter notebooks.
```

File: content/en/users/tutorials/jupyter-datahub-virtual-machine/_index.md
Comment (sebastian-luna-valero) [inline content/en/users/tutorials/jupyter-datahub-virtual-machine/_index.md:98]: No need to specify an environment `name`?

**Review by sebastian-luna-valero — COMMENTED** [review]
Great work!

Eager to test this out!

File: content/en/users/tutorials/jupyter-datahub-virtual-machine/_index.md
Comment (enolfc) [reply content/en/users/tutorials/jupyter-datahub-virtual-machine/_index.md:98]: it's the default one, so no need to have a name (at least it was working for me)

Comment (sebastian-luna-valero) [reply content/en/users/tutorials/jupyter-datahub-virtual-machine/_index.md:98]: alright, thanks!

**Comment by gwarf** [comment]
@thebe14 would you be able to provide feedback on this one? Or should it be done by someone else?

File: content/en/users/tutorials/jupyter-datahub-virtual-machine/_index.md
Comment (thebe14) [inline content/en/users/tutorials/jupyter-datahub-virtual-machine/_index.md:17]: I would keep the title as just "Requirements"

**Comment by thebe14** [comment]
LFTM. However, I would like to see a uniform structure for all tutorials. It makes the right-hand-side navigation much easier to follow.

```markdown
## Overview
## Requirements
## Major goal 1
### Step 1
### Step 2
...
## Major goal 2
### Step 1
...
```

**Review by thebe14 — APPROVED** [review]

**Comment by gwarf** [comment]
@thebe14 I'm confused by your review: should change be made or can we merge this PR as it is?

**Comment by andrea-manzi** [comment]
i would like to test it in Fedcloud:-) is the new image  now available in the vo.access.egi.eu VO?

**Comment by thebe14** [comment]
> @thebe14 I'm confused by your review: should change be made or can we merge this PR as it is?

I only suggested a minor change to the second title, but I will not insist on it. It can be merged. Maybe I'll do a PR later to try to uniformize the structure of all tutorials.

**Comment by gwarf** [comment]
Thanks for the clarification @thebe14!
@enolfc please look at the open conversation, @andrea-manzi please let us know if we should wait for your review before merging.

File: content/en/users/tutorials/jupyter-datahub-virtual-machine/_index.md
Comment (enolfc) [reply content/en/users/tutorials/jupyter-datahub-virtual-machine/_index.md:17]: Fixed

**Comment by enolfc** [comment]
> i would like to test it in Fedcloud:-) is the new image now available in the vo.access.egi.eu VO?

Image added to vo.access.egi.eu, should start to get propagated soon

**Comment by enolfc** [comment]
> LGTM. However, I would like to see a uniform structure for all tutorials. It makes the right-hand-side navigation much easier to follow.
> 
> ```gfm
> ## Overview
> ## Requirements
> ## Major goal 1
> ### Step 1
> ### Step 2
> ...
> ## Major goal 2
> ### Step 1
> ...
> ```

Reorganised to match that structure

File: content/en/users/tutorials/jupyter-datahub-virtual-machine/_index.md
Comment (sebastian-luna-valero) [inline content/en/users/tutorials/jupyter-datahub-virtual-machine/_index.md:13]:
```suggestion
that runs a browser-accessible jupyter server with DataHub spaces mounted. This
```

File: content/en/users/tutorials/jupyter-datahub-virtual-machine/_index.md
Comment (sebastian-luna-valero) [inline content/en/users/tutorials/jupyter-datahub-virtual-machine/_index.md:41]:
```suggestion
[here how to prepare it](https://superuser.com/a/853957).
```

File: content/en/users/tutorials/jupyter-datahub-virtual-machine/_index.md
Comment (sebastian-luna-valero) [inline content/en/users/tutorials/jupyter-datahub-virtual-machine/_index.md:67]:
```suggestion
And create the `mount.sh` script by adding the `ONECLIENT_ACCESS_TOKEN` and
```

File: content/en/users/tutorials/jupyter-datahub-virtual-machine/_index.md
Comment (sebastian-luna-valero) [inline content/en/users/tutorials/jupyter-datahub-virtual-machine/_index.md:84]:
```suggestion
$ bash mount.sh
```

File: content/en/users/tutorials/jupyter-datahub-virtual-machine/_index.md
Comment (sebastian-luna-valero) [inline content/en/users/tutorials/jupyter-datahub-virtual-machine/_index.md:87]: `/mnt/datahub` has been configured for the `ubuntu` user.
I have used IM, which creates the `cloudadm` user and I get:
```
fusermount: user has no write access to mountpoint /mnt/datahub
```
Currently the manual solution is:
```
sudo chown cloudadm.cloudadm -R /mnt/datahub/
```

**Comment by sebastian-luna-valero** [comment]
When I try accessing Jupyter with `http://<floating-ip/` I get:
image

File: content/en/users/tutorials/jupyter-datahub-virtual-machine/_index.md
Comment (enolfc) [reply content/en/users/tutorials/jupyter-datahub-virtual-machine/_index.md:84]: This needs to be fixed in the image

File: content/en/users/tutorials/jupyter-datahub-virtual-machine/_index.md
Comment (enolfc) [reply content/en/users/tutorials/jupyter-datahub-virtual-machine/_index.md:87]: I think I will just use jovyan as user for everything
Was trying to avoid root access as much as possible

**Comment by enolfc** [comment]
New version of the VM (and instructions!) are now in vo.access.egi.eu list. Give it a few hours to sync across sites

File: content/en/users/tutorials/jupyter-datahub-virtual-machine/_index.md
Comment (enolfc) [reply content/en/users/tutorials/jupyter-datahub-virtual-machine/_index.md]: Mounting now done as `jovyan`, problem should be gone

**Comment by sebastian-luna-valero** [comment]
It works great, thanks!

**Comment by gwarf** [comment]
So all good, it can now be merged?

File: content/en/users/tutorials/jupyter-datahub-virtual-machine/_index.md
Comment (andrea-manzi) [inline content/en/users/tutorials/jupyter-datahub-virtual-machine/_index.md:13]:
```suggestion
that runs a browser-accessible Jupiter server with DataHub spaces mounted. This
```

File: content/en/users/tutorials/jupyter-datahub-virtual-machine/_index.md
Comment (andrea-manzi) [inline content/en/users/tutorials/jupyter-datahub-virtual-machine/_index.md:51]:
```suggestion
### Step 2: Get your token for the Jupiter server
```

File: content/en/users/tutorials/jupyter-datahub-virtual-machine/_index.md
Comment (andrea-manzi) [inline content/en/users/tutorials/jupyter-datahub-virtual-machine/_index.md:74]:
```suggestion
Jupiter or via ssh. For example for installing `tensorflow`:
```

Comment (andrea-manzi) [inline content/en/users/tutorials/jupyter-datahub-virtual-machine/_index.md:74]:
```suggestion
Jupyter or via ssh. For example for installing `tensorflow`:
```

File: content/en/users/tutorials/jupyter-datahub-virtual-machine/_index.md
Comment (andrea-manzi) [inline content/en/users/tutorials/jupyter-datahub-virtual-machine/_index.md:87]:
```suggestion
You will also need the IP or address of your closest Oneprovider for the spaces you
```

File: content/en/users/tutorials/jupyter-datahub-virtual-machine/_index.md
Comment (gwarf) [reply content/en/users/tutorials/jupyter-datahub-virtual-machine/_index.md:51]: Oh, maybe you meant to add an upper case first letter and made a typo/automatic correction of the y/i

Comment (andrea-manzi) [inline content/en/users/tutorials/jupyter-datahub-virtual-machine/_index.md:51]:
```suggestion
### Step 2: Get your token for the Jupyter server
```

**Review by andrea-manzi — APPROVED** [review]
Looks good to me, @sebastian-luna-valero tested it so we are good to go:-)

**Comment by gwarf** [comment]
Thanks guys!
Let's merge once all tests have passed.

**Pull request merged**
